src/ServiceManagement/Services/Commands.Utilities/Websites/WebsitesClient.cs

```diff
@@ -1360,11 +1361,50 @@ private DeploymentBaseOptions CreateRemoteDeploymentBaseOptions(string websiteNa
                 AuthenticationType = "Basic",
                 TempAgent = false
             };
+            
+            var azureSessionUserAgent = GetDeploymentBaseOptionsUserAgent();
+            if (!string.IsNullOrEmpty(azureSessionUserAgent))
+            {
+                remoteBaseOptions.UserAgent = azureSessionUserAgent;
+            }
 
             return remoteBaseOptions;
         }
 
         /// <summary>
+        /// Gets remote deployment base options useragent using AzureSession/WebSiteManagementClient.
+        /// </summary>
+        /// <returns>useragent string.</returns>
+        private string GetDeploymentBaseOptionsUserAgent()
+        {
+            var userAgent = string.Empty;
+            var managementClient = this.WebsiteManagementClient as WebSiteManagementClient;
+            var userAgents = AzureSession.ClientFactory.UserAgents.Any()
```

Comment:
No, just use the UserAgents from the management client - they are automatically propagated there from the ClientFactory when you create the client.  The code that checks the ClientFactory UserAgents is redeundant